guppylang/compiler/cfg_compiler.py

```diff
@@ -92,7 +91,11 @@ def compile_bb(
                 graph=graph,
                 unit_sum=branch_port,
                 output_vars=[
-                    [v for v in row if not v.ty.linear or is_return_var(v.name)]
+                    [
+                        v
+                        for v in sort_vars(row)
+                        if not v.ty.linear or is_return_var(v.name)
+                    ]
```

Comment:
I moved the call to `sort_vars` from `choose_vars_for_tuple_sum()` below to here as a drive-by since this is more consistent with the other sorting logic in this method